lms/djangoapps/bulk_email/models.py

```diff
@@ -70,7 +70,21 @@ class Meta(object):
         app_label = "bulk_email"
 
     def __unicode__(self):
-        return "CourseEmail Target for: {}".format(self.target_type)
+        return "CourseEmail Target: {}".format(self.short_display())
+
+    def short_display(self):
+        # django polymorphism is very odd, and doesn't work automatically
```

Comment:
To summarize my findings on this issue:

- Django models don't necessarily handle polymorphism like normal python classes, due to black magic on the backend that separates them into different tables
- Because of this, a `Target.short_display()` call will hit this code, not the child class as you'd expect
- We can access the child method, if we know it exists, by using `self.cohorttarget` (Note that `cohorttarget` is the name of the table where these models actually store their data)